cfme/networks/topology.py

```diff
@@ -0,0 +1,244 @@
+class TopologyLine(BaseEntity):
+    """ Class representing line connecting 2 nodes in Topology
+        Consists of Browser element
+    """
+    element = attr.ib()
+
+
+@attr.s
+class TopologyLineCollection(BaseCollection):
+    """Collection object for lines in topology"""
+    ENTITY = TopologyLine
+
+    def instantiate(self, *args, **kwargs):
+        line = super(TopologyLineCollection, self).instantiate(*args, **kwargs)
+        assert line.element, 'Element should not be None'
+        line.connection = line.element.get_attribute('class')
+        line.x1 = round(float(line.element.get_attribute('x1')), 1)
+        line.x2 = round(float(line.element.get_attribute('x2')), 1)
+        line.y1 = round(float(line.element.get_attribute('y1')), 1)
+        line.y2 = round(float(line.element.get_attribute('y2')), 1)
+        return line
+
+    def filter(self, topology):
+        lines = topology.browser.elements(topology.view.LINES)
+        return [self.instantiate(element=line) for line in lines]
+
+
+@attr.s
+class Topology(BaseEntity):
+    "Class represents SDN topology"
+    elements_obj = attr.ib(default=None)
+    lines_obj = attr.ib(default=None)
+    legends_obj = attr.ib(default=None)
+    elements_col = attr.ib(default=None)
+    lines_col = attr.ib(default=None)
+    legends_col = attr.ib(default=None)
+    element_ref = attr.ib(default=None)
+    display_names = attr.ib(default=None)
+    view = attr.ib(default=None)
+
+    def reload_elements_and_lines(self):
+        self.elements_obj = []
+        self.lines_obj = self.lines_col.filter(self)
+        found_elements = self.elements_col.filter(self)
+
+        if found_elements:
+            self.element_ref = found_elements[-1]
+            wait_for(lambda: self.movement_stopped, delay=2, num_sec=30)
+            self.elements_obj = self.elements_col.filter(self)
+
+    def reload_legends(self):
+        self.legends_obj = self.legends_col.filter(self)
+        element = self.browser.element(self.view.DISPLAY_NAME)
+        self.display_names = TopologyDisplayNames(self.appliance, element)
+
+    @property
+    def refresh(self):
+        self.view = navigate_to(self, 'All')
+        self.reload_elements_and_lines()
+        self.reload_legends()
+
+    @property
+    def movement_stopped(self):
+        element = self.elements_col.filter(self)[-1]
+        if element.x == self.element_ref.x and element.y == self.element_ref.y:
+            return True
+        self.element_ref = element
+        return False
+
+    @property
+    def legends_collection(self):
+        return self.legends_col
+
+    @property
+    def elements_collection(self):
+        return self.elements_col
+
+    @property
+    def lines_collection(self):
+        return self.lines_col
+
+    @property
+    def legends(self):
+        return self.legends_obj
+
+    @property
+    def elements(self):
+        return self.elements_obj
+
+    @property
+    def lines(self):
+        return self.lines_obj
```

Comment:
**REQUIRED**
These three properties are not needed. Just rename `legends_obj` and others attribute to `legends` and so on.